src/modules/generateRandomNumber.js

```diff
@@ -7,7 +7,21 @@
  * @return {number} A random 4-digit number
  */
 function generateRandomNumber() {
-  /* Write your code here */
+  let digits = ['1', '2', '3', '4', '5', '6', '7', '8', '9'];
+  let number = digits[Math.floor(Math.random() * digits.length)];
+
+  digits = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'].filter(
+    (d) => !number.includes(d),
+  );
+
+  while (number.length < 4) {
+    const randomIndex = Math.floor(Math.random() * digits.length);
+
+    number += digits[randomIndex];
+    digits.splice(randomIndex, 1);
+  }
+
+  return Number(number);
```

Comment:
The function `generateRandomNumber` generates a random number with unique digits, which is good. However, the current implementation has a potential issue. It always generates a number starting with a digit between 1 and 9, and then includes 0 in the subsequent digits. If the game rules allow the first digit to be 0, this function would not comply with the rules. Ensure the game rules are being followed correctly.